bin/drop_sample_annot.py

```diff
@@ -8,36 +8,40 @@
 import os
 import pandas as pd
 
+SAMPLE_ANNOTATION_COLUMNS = [
+    "RNA_ID",
+    "RNA_BAM_FILE",
+    "DNA_VCF_FILE",
+    "DNA_ID",
+    "DROP_GROUP",
+    "PAIRED_END",
+    "COUNT_MODE",
+    "COUNT_OVERLAPS",
+    "SPLICE_COUNTS_DIR",
+    "STRAND",
+    "HPO_TERMS",
+    "GENE_COUNTS_FILE",
+    "GENE_ANNOTATION",
+    "GENOME",
+]
+
 
 class SampleAnnotation:
     """SampleAnnotation class"""
 
-    SAMPLE_ANNOTATION_COLUMNS = [
-        "RNA_ID",
-        "RNA_BAM_FILE",
-        "DNA_VCF_FILE",
-        "DNA_ID",
-        "DROP_GROUP",
-        "PAIRED_END",
-        "COUNT_MODE",
-        "COUNT_OVERLAPS",
-        "SPLICE_COUNTS_DIR",
-        "STRAND",
-        "HPO_TERMS",
-        "GENE_COUNTS_FILE",
-        "GENE_ANNOTATION",
-        "GENOME",
-    ]
-
-    def __init__(self, bam, sample, strandedness, single_end, gtf, count_file, out_file):
-        """Write the Sample Annotation tsv file"""
+    SAMPLE_ANNOTATION_COLUMNS = SAMPLE_ANNOTATION_COLUMNS
+
+    def __init__(
+        self, bam: str, samples: str, strandedness: str, single_end: str, gtf: str, count_file: str, out_file: str
+    ):
+        """
+        Write the Sample Annotation tsv file.
+        """
         with open(out_file, "w") as tsv_file:
             fieldnames = self.SAMPLE_ANNOTATION_COLUMNS
             writer = csv.DictWriter(tsv_file, fieldnames=fieldnames, delimiter="\t")
-
             writer.writeheader()
-
-            for index, id in enumerate(sample):
+            for index, id in enumerate(samples):
                 sa_dict = {}.fromkeys(fieldnames, "NA")
                 sa_dict["RNA_ID"] = id
                 sa_dict["DROP_GROUP"] = "outrider,fraser"
```

Comment:
typheint `paired_end_func`

Reply:
I have made it into a separate function for readability